ext/src/zpages/tracez_processor.cc

```diff
@@ -0,0 +1,37 @@
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext {
+namespace zpages {
+
+  void TracezSpanProcessor::OnStart(opentelemetry::sdk::trace::Recordable &span) noexcept {
+    if (shutdown_signal_received_) return;
+    spans_.running.insert(static_cast<opentelemetry::sdk::trace::SpanData*>(&span));
+  }
+
+  void TracezSpanProcessor::OnEnd(std::unique_ptr<opentelemetry::sdk::trace::Recordable> &&span) noexcept {
+     if (shutdown_signal_received_ || span == nullptr) return;
+     auto completed_span_raw = static_cast<opentelemetry::sdk::trace::SpanData*>(span.release());
+     auto completed_span_it = spans_.running.find(completed_span_raw);
+     if (completed_span_it != spans_.running.end()) {
```

Comment:
In case this condition returns false, we'd leak `completed_span_raw`.

Reply:
To resolve that issue, I'm thinking `completed_span_raw` should be deleted when the condition returns false then? Would adding the following line be acceptable:
``` 
...
} else delete complete_span_raw;
..
```
Also, in the case that this returns false somehow, my understanding is that we don't want to add that span to completed; does this sound appropriate?

Reply:
You can also just get the pointer first, without releasing:
```c
auto completed_span_raw = static_cast<opentelemetry::sdk::trace::SpanData*>(span.get());
```
And then do the release when you finally transfer ownership:
```c
spans_.completed.push_back(
           std::unique_ptr<opentelemetry::sdk::trace::SpanData>(span.release()));
```
That's more in the spirit of smart pointers.

Reply:
Thank you, I’ll be sure to change that!